Mage.Sets/src/mage/cards/k/KharnTheBetrayer.java

```diff
@@ -0,0 +1,145 @@
+import mage.constants.SuperType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.target.TargetPlayer;
+import mage.target.common.TargetOpponent;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ * @author Xanderhall
+ */
+public class KharnTheBetrayer extends CardImpl {
+
+    public KharnTheBetrayer(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{R}");
+        this.addSuperType(SuperType.LEGENDARY);
+        this.addSubType(SubType.ASTARTES, SubType.BERSERKER);
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(1);
+
+        // {this} attacks or blocks each combat if able
+        Ability ability = new SimpleStaticAbility(new AttacksIfAbleSourceEffect(Duration.WhileOnBattlefield).setText("{this} attacks"));
+        ability.addEffect(new BlocksIfAbleSourceEffect(Duration.WhileOnBattlefield).setText("blocks each combat if able").concatBy("or"));
+        this.addAbility(ability.withFlavorWord("Berzerker"));
+
+        // When you lose control of {this}, draw two cards
+        this.addAbility(new KharnTheBetrayerTriggeredAbility().withFlavorWord("Sigil of Corruption"));
+
+        // If damage would be dealt to {this}, prevent that damage and an opponent of your choice gains control of it.
+        Ability replacementAbility = new SimpleStaticAbility(new KharnTheBetrayerEffect());
+        this.addAbility(replacementAbility.withFlavorWord("The Betrayer"));
+    }
+
+    protected KharnTheBetrayer(final KharnTheBetrayer card) {
+        super(card);
+    }
+
+    @Override
+    public Card copy() {
+        return new KharnTheBetrayer(this);
+    }
+    
+}
+
+class KharnTheBetrayerEffect extends ReplacementEffectImpl {
+
+    public KharnTheBetrayerEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.PreventDamage);
+        this.staticText = "If damage would be dealt to {this}, prevent that damage and an opponent of your choice gains control of it.";
+    }
+
+    protected KharnTheBetrayerEffect(final KharnTheBetrayerEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        int damage = game.preventDamage(event, source, game, Integer.MAX_VALUE).getPreventedDamage();
+        Player player = game.getPlayer(source.getControllerId());
+
+        if (player != null && damage > 0) {
+            TargetPlayer target = new TargetOpponent();
+            target.setNotTarget(true);
+            if (!player.choose(outcome, target, source, game)) {
+                return false;
+            }
+            ContinuousEffect effect = new GainControlTargetEffect(Duration.Custom, true, target.getFirstTarget());
+            effect.setTargetPointer(new FixedTarget(source.getSourceId(), game));
+            
+            ReflexiveTriggeredAbility ability 
```

Comment:
I don't think there's a reflexive trigger at all here. Rather than a reflexive triggered ability, you just want `game.addEffect(effect, source)`